Indicators/RelativeVigorIndex.cs

```diff
@@ -0,0 +1,133 @@
+        /// </summary>
+        public RollingWindow<TradeBar> Bars { get; set; }
+
+        /// <summary>
+        /// Creates the indicator
+        /// </summary>
+        /// <param name="name">string - the name of the indicator</param>
+        /// <param name="period">int - half the number of periods in the dominant market cycle. </param>
+        public RelativeVigorIndex(string name, int period)
+            : base(name)
+        {
+
+            value1 = new RollingWindow<IndicatorDataPoint>(period);
+            value2 = new RollingWindow<IndicatorDataPoint>(period);
+            RviWindow = new RollingWindow<IndicatorDataPoint>(period);
+            Bars = new RollingWindow<TradeBar>(period);
+        }
+
+
+        /// <summary>
+        /// Automatically names the indicator
+        /// </summary>
+        /// <param name="period">int - the length of computation cycle</param>
+        public RelativeVigorIndex(int period)
+            : this("RVI_" + period, period)
+        {
+        }
+
+        /// <summary>
+        /// Gets a flag indicating when this indicator is ready and fully initialized
+        /// </summary>
+        public override bool IsReady
+        {
+            get { return value1.IsReady && value2.IsReady && RviWindow.IsReady; }
+        }
+        /// <summary>
+        /// Computes the next value of this indicator from the given state
+        /// </summary>
+        /// <param name="input">The input given to the indicator</param>
+        /// <returns>A new value for this indicator</returns>
+        protected override decimal ComputeNextValue(TradeBar input)
+        {
+            // Save the bar for summing
+            Bars.Add(input);
+
+            // check if it is not ready add a 0 to the RviWindow
+            if (!Bars.IsReady)
+            {
+                RviWindow.Add(new IndicatorDataPoint(input.EndTime, 0.0m));
+            }
+            else
+            {
+                // Bars have been added, so Bars[0] is the current value of the input
+
+
+                // Filter the Close - Open with a four-bar symetical FIR filter before the terms are summed.
+                var v1 = ((Bars[0].Close - Bars[0].Open) + 2 * (Bars[1].Close - Bars[1].Open) +
+                          2 * (Bars[2].Close - Bars[3].Open) + (Bars[3].Close - Bars[3].Open) / 6);
+                value1.Add(new IndicatorDataPoint(input.EndTime, v1));
+
+                // Filter the High - Low with a four-bar symetical FIR filter before the terms are summed.
+                var v2 = ((Bars[0].High - Bars[0].Low) + 2 * (Bars[1].High - Bars[1].Low) +
+                          2 * (Bars[2].High - Bars[3].Low) + (Bars[3].High - Bars[3].Low) / 6);
+                value2.Add(new IndicatorDataPoint(input.EndTime, v2));
+
+                // The numerator and denominator are summed independently
+                decimal num = value1.Sum(point => point.Value);
+                decimal denom = value2.Sum(point => point.Value);
```

Comment:
It would be more performant if we used the Sum indicator type here so we don't need to recompute the whole thing each time.